File: src/1-500/006/ZigZagConversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if not (numRows-1):
            return s
        conversion = ""
        for i in range(numRows):
            rowstr = ""
            index = i
            flag = 0
            while (index < len(s)):
                increment = i * 2 if flag else (numRows-i-1) * 2
                rowstr = rowstr + s[index] if increment else rowstr
                index = index + increment
                flag = not flag
            conversion = conversion + rowstr
        return conversion

File: src/1-500/006/ZigZagConversion.java
class Solution {
    public String convert(String s, int numRows) {
		if (s==null || s.isEmpty() || numRows <= 1 || s.length() <= numRows) return s;
		
		int len = s.length();
		int lenCycle = 2 * numRows - 2;
		StringBuilder res = new StringBuilder();

		for (int i=0; i< numRows; i++) {
			for (int j=0; j+i<len; j+=lenCycle) {
				res.append(s.charAt(j+i));
				if (i!=0 && i!=numRows-1 && j+lenCycle-i<len) {
					res.append(s.charAt(j+lenCycle-i));
				}
			}			
		}
		
		return res.toString();
	}
}